denops/@denops-private/denops.ts

```diff
@@ -72,7 +76,20 @@ export class DenopsImpl implements Denops {
     fn: string,
     ...args: unknown[]
   ): Promise<unknown> {
-    return this.#service.dispatch(name, fn, args);
+    if (this.#service.loaded(name)) {
+      return this.#service.dispatch(name, fn, args);
+    }
+    return new Promise((resolve, reject) => {
+      const callback = (ev: Event) => {
+        if (name === maybe(ev, isPluginLoadEndEvent)?.detail?.name) {
+          globalThis.removeEventListener("denops:pluginloadend", callback);
+          this.#service.dispatch(name, fn, args)
+            .then((v) => resolve(v))
+            .catch((err) => reject(err));
+        }
+      };
+      globalThis.addEventListener("denops:pluginloadend", callback);
```

Comment:
Make sure to reject the promise when handling `denops:pluginloadfail`.
I suggest using `AbortSignal` instead of `removeEventListener` to remove multiple handlers at the same time.